package/MDAnalysis/analysis/lineardensity.py

```diff
@@ -0,0 +1,146 @@
+            positions = np.array([elem.centroid() for elem in self.group]) # COM for res/frag/etc
+
+        for dim in ['x', 'y', 'z']:
+            idx = self.results[dim]['dim']
+
+            key = 'pos'
+            key_std = 'pos_std'
+            # histogram for positions weighted on masses
+            hist, _ = np.histogram(positions[:, idx], weights=self.masses,
+                                   bins=self.nbins, range=(0.0, max(self.dimensions)))
+
+            self.results[dim][key] = np.sum([self.results[dim][key], hist], axis=0)
+            self.results[dim][key_std] = np.sum([self.results[dim][key_std],
+                                                 np.square(hist)], axis=0)
+
+            key = 'char'
+            key_std = 'char_std'
+            # histogram for positions weighted on charges
+            hist, _ = np.histogram(positions[:, idx], weights=self.charges,
+                                   bins=self.nbins, range=(0.0, max(self.dimensions)))
+
+            self.results[dim][key] = np.sum([self.results[dim][key], hist], axis=0)
+            self.results[dim][key_std] = np.sum([self.results[dim][key_std], np.square(hist)],
+                                                axis=0)
+
+    def _conclude(self):
+        k = 6.022e-1 # divide by avodagro and convert from A3 to cm3
+        bins = np.linspace(0.0, max(self.dimensions), num=self.nbins)
+
+        # Average results over the  number of configurations
+        for dim in ['x', 'y', 'z']:
+            for key in ['pos', 'pos_std', 'char', 'char_std']:
+                self.results[dim][key] /= self.nframes
+            # Computed standard deviation for the error
+            self.results[dim]['pos_std'] = np.sqrt(self.results[dim]['pos_std']
+                                                   - np.square(self.results[dim]['pos']))
+            self.results[dim]['char_std'] = np.sqrt(self.results[dim]['char_std']
+                                                    - np.square(self.results[dim]['char']))
+
+
+        # Create list of results which will be output
+        output = [bins]
+
+        for dim in ['x', 'y', 'z']:
+            output.append(self.results[dim]['pos']/(self.results[dim]['slice volume']*k))
+            output.append(self.results[dim]['pos_std']/(self.results[dim]['slice volume']*k))
+
+        for dim in ['x', 'y', 'z']:
+            output.append(self.results[dim]['char']/(self.results[dim]['slice volume']*k))
+            output.append(self.results[dim]['char_std']/(self.results[dim]['slice volume']*k))
+
+        # Define filename and write to output
+        filename = self.trajname+"."+self.description+".ldens"
+        density = self.totalmass/self.volume
+        header = "1 coord [Ang] 2-7 mass density (x,sx,y,sz,z,sz) [g/cm^3]" + \
+            "8-13 charge density (x,sx,y,sz,z,sz) [e/A^3]\n Average density: "\
+            +str(density)+" g/cm3"
+        np.savetxt(filename, np.column_stack(output), header=header)
```

Comment:
put saving into an optional `save` method.  Also, maybe an option between `savetxt` and `savez` for ascii/not ascii